api/kv_test.go

```diff
@@ -0,0 +1,120 @@
+package api
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestExtractVersionMetadata(t *testing.T) {
```

Comment:
This is a great test! I think the other extract methods could benefit from testing as well. Maybe `Example_` tests for the path methods.